src/analytics/baseTracker/baseTracker.ts

```diff
@@ -63,4 +63,27 @@ export class BaseTracker {
       ? document.referrer?.toLowerCase()
       : "undefined";
   }
+
+  /**
+   * Retrieves the label of a field.
+   *
+   * @return {string} The label of the field.
+   */
+  getFieldLabel(): string {
+    let labels: HTMLCollectionOf<HTMLLegendElement | HTMLLabelElement> =
+      document.getElementsByTagName("legend");
+    if (!labels.length) {
+      labels = document.getElementsByTagName("label");
+    }
```

Comment:
Will there ever be an instance where both legends and labels are present on the page?

Reply:
Not in what I have seen. 
Priority is to take the legend if we have one. And after the label if no legend.
In gov uk frontend, legend tag is used to show the "question/purpose" of the fields. Always use for radio buttons and checboxes. Not often for dropdown or free text (but still possible to put one).